### ISSUE TITLE
Change account for a single request

### ISSUE THREAD

**Issue by LowieDelneste** [code4me/4me-sdk-graphql-dotnet#7]
Hello

In our application we want to have the option to change the AccountId (x-4me-account header) for a single request. Currently this is only possible at client level. As I see it (please correct me if I'm wrong), the client is meant to be only created once (due to the way the [HttpClient](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-net-http-httpclient#instancing) is used. [See code](https://github.com/code4me/4me-sdk-graphql-dotnet/blob/6e0fe5a5f7322c5e77a3cc08265fef242f0376fc/Scr/Sdk4me.GraphQL/Sdk4meClient.cs#L182))? Because of this, when multiple requests are executed at the same time, all of them have the same (possible changed) account header. 

Can the API be modified so that we can pass an AccountId with the `client.Get(Query)` method?
A quick code example to explain what I mean can be found [here](https://github.com/LowieDelneste/4me-sdk-graphql-dotnet/commit/748a08c524c2c1c1442755268b1ad35f785c6e4f ).

This can even be added to the Mutation method. 

Just an extra thought: maybe it is better to create a client per account?

Thanks in advance!

**Comment by klaasvandeweerdt** [comment]
Hi @LowieDelneste,

Yes, the client is only created once, and the Account ID is stored on client level.
Thanks for the example. I think both make sense, it could even be an optional argument, if `null` it takes the Account ID stored on client level. Nevertheless, it will also add a layer of complexity, as authentication tokens are stored on client level as well.
I don’t know how many 4me accounts you are connecting to but getting it in a Dictionary could help.

``` csharp
Dictionary<string, Sdk4meClient> clients = new()
{
    { "accountID-1", new Sdk4meClient(new AuthenticationTokenCollection() { }, "accountID-1", EnvironmentType.Production, EnvironmentRegion.EU) },
    { "accountID-1", new Sdk4meClient(new AuthenticationTokenCollection() { }, "accountID-2", EnvironmentType.Production, EnvironmentRegion.EU) },
    { "accountID-2", new Sdk4meClient(new AuthenticationTokenCollection() { }, "accountID-3", EnvironmentType.Production, EnvironmentRegion.EU) }
};

var result = clients["account-1"].Get(new AccountQuery().Select(AccountField.ID)).Result;
```

**Comment by LowieDelneste** [comment]
Hi @klaasvandeweerdt

Thanks for the reply.
Yeah, the solution with the dictionary is what I came up with after asking this question. :) Maybe this is something that can be mentioned in the documentation?

**Comment by klaasvandeweerdt** [comment]
Hi @LowieDelneste,
I have done some tests with a few different scenario's, but I think that it makes it to complex, and sometimes quite hard to troubleshoot. I had it running against QA, and two demo instance, it didn't took me long before having the wrong tokens on the wrong connection. Another idea could be a custom Dictionary in the SDK, but the only benefit I see until now entering one time the account ID instead of two.
I will push an update this weekend based on the latest schema, and include readme update with the dictionary recommendation when running multiple requests to multiple account at the same time.

**Comment by klaasvandeweerdt** [comment]
Hi @LowieDelneste,
I've uploaded the most recent version to Git, without any specific changes for this issue. I've included the suggested `Dictionary` method in the documentation. Additionally, I've released a new NuGet version, 1.2.1, since the previous one was two months old.

**Comment by klaasvandeweerdt** [comment]
Hi @LowieDelneste,
Eventually, I modify a section of the code to ensure consistency of the `AccountID` across recursive requests and in the trace logs. However, this does not solve the challenge of multiple simultaneous requests over a single `Skd4meClient` instance.
